packages/base-driver/lib/protocol/protocol.js

```diff
@@ -283,12 +291,15 @@ function buildHandler (app, method, path, spec, driver, isSessCmd) {
         `${driver.constructor.name}.${spec.command}() with args: ` +
         _.truncate(JSON.stringify(args), {length: MAX_LOG_BODY_LENGTH}));
 
-      if (driver.executeCommand) {
-        driverRes = await driver.executeCommand(spec.command, ...args);
-      } else {
-        driverRes = await driver.execute(spec.command, ...args);
+      if (pluginOverrodeProxy) {
```

Comment:
What's the most direct way of understanding wtf a JSONWP server is and why we'd proxy commands to it?

Reply:
JSONWP is just the old name of the WebDriver protocol. long ago we created the jsonwp-proxy package which allows us to proxy webdriver commands to another webdriver compatible server. the jsonwp protocol itself is defunct, but the package has been updated to use the w3c webdriver protocol.

so the strings "jsonwp" or "jwp" (same thing) are still in our code and our mostly vestigial. at some point we need to go through and actually remove all efforts to handle the (actual) jsonwp, and at that point could rename everything that's left as "wd" or something, so "wdproxy" instead of "jwproxy" or whatever.

Reply:
but to answer 'why we'd proxy commands to it', many of our drivers are set up as a node.js shim which talks to a webdriver server on the device (e.g. WDA or the UiAuto 2 server). because we have this proxy mode, it means we don't have to write any code at all in the node.js portion of the driver and commands will be transparently proxied to the server on the device).

Reply:
the server on the device... is this natively supported, or?

Reply:
no. by "server on the device" i mean "some software we wrote that runs on a device, usually a server, usually launched inside a test case because what the device thinks we are doing is running an XCTest or a UiAutomator test". for an example of "server on the device" see WebDriverAgent. "server on the device" is exactly what it would be great if Apple and Google made instead of us.